runelite-client/src/test/java/net/runelite/client/plugins/skillcalculator/CalculatorTypeTest.java

```diff
@@ -30,21 +30,32 @@
 
 public class CalculatorTypeTest
 {
+
 	@Test
-	public void skillActionsInLevelOrder()
+	public void skillActionsInOrder()
 	{
 		for (final CalculatorType calculatorType : CalculatorType.values())
 		{
-			int level = 1;
+			int prevLevel = 0;
+			float prevXP = 0.0f;
+			String prevName = "";
 
 			for (final SkillAction skillAction : calculatorType.getSkillActions())
 			{
-				if (skillAction.getLevel() < level)
+				int currentLevel = skillAction.getLevel();
+				float currentXP = skillAction.getXp();
+				String currentName = skillAction.toString();
+
+				if (currentLevel < prevLevel ||
+						(currentLevel == prevLevel && currentXP < prevXP) ||
+						(currentLevel == prevLevel && currentXP == prevXP && currentName.compareTo(prevName) < 0))
 				{
 					fail("Skill action " + skillAction + " is out of order for " + calculatorType.getSkill().getName());
 				}
```

Comment:
Let's use different failure messages for each condition which can fail; after all, these are here so that newer contributors can understand what went wrong.